applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt_adv.c

```diff
@@ -0,0 +1,296 @@
+			return;
+		}
+	}
+
+	if (ext_adv_set != NULL) {
+		ext_adv_set(ext_adv);
+	}
+
+	LOG_INF("Advertising successfully started");
+}
+
+static void connected_cb(struct bt_conn *conn, uint8_t err)
+{
+	int ret;
+	char addr[BT_ADDR_LE_STR_LEN];
+	uint16_t conn_handle;
+	enum ble_hci_vs_tx_power conn_tx_pwr;
+
+	if (err) {
+		return;
+	}
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+	LOG_INF("Connected: %s", addr);
+
+	ret = bt_hci_get_conn_handle(conn, &conn_handle);
+	if (ret) {
+		LOG_ERR("Unable to get conn handle");
+	} else {
+#if (CONFIG_NRF_21540_ACTIVE)
+		conn_tx_pwr = CONFIG_NRF_21540_MAIN_DBM;
+#else
+		conn_tx_pwr = CONFIG_BLE_CONN_TX_POWER_DBM;
+#endif /* (CONFIG_NRF_21540_ACTIVE) */
+		ret = ble_hci_vsc_conn_tx_pwr_set(conn_handle, conn_tx_pwr);
+		if (ret) {
+			LOG_ERR("Failed to set TX power for conn");
+		} else {
+			LOG_DBG("TX power set to %d dBm for connection %p", conn_tx_pwr,
+				(void *)conn);
+		}
+	}
+
+	if (conn_set != NULL) {
+		conn_set(bt_conn_ref(conn));
+	}
+}
+
+static void disconnected_cb(struct bt_conn *conn, uint8_t reason)
+{
+	int ret;
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_INF("Disconnected: %s (reason 0x%02x)", addr, reason);
+
+	bt_conn_unref(conn);
+
+	if (conn_set != NULL) {
+		conn_set(NULL);
+	}
+
+	if (IS_ENABLED(CONFIG_BT_MCC)) {
+		ret = ble_mcp_conn_disconnected(conn);
+		if (ret) {
+			LOG_ERR("ble_msc_conn_disconnected failed with %d", ret);
+		}
+	}
+
+	k_work_submit(&adv_work);
+}
+
+#if defined(CONFIG_BT_SMP)
+static void security_changed_cb(struct bt_conn *conn, bt_security_t level, enum bt_security_err err)
+{
+	int ret;
+
+	if (err) {
+		LOG_ERR("Security failed: level %d err %d", level, err);
+		ret = bt_conn_disconnect(conn, err);
+		if (ret) {
+			LOG_ERR("Failed to disconnect %d", ret);
+		}
+	} else {
+		LOG_DBG("Security changed: level %d", level);
+	}
+}
+#endif /* defined(CONFIG_BT_SMP) */
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected_cb,
+	.disconnected = disconnected_cb,
+#if defined(CONFIG_BT_SMP)
+	.security_changed = security_changed_cb,
+#endif /* defined(CONFIG_BT_SMP) */
+};
+
+int bt_mgmt_adv_start(const struct bt_data *adv, size_t adv_size, const struct bt_data *per_adv,
+		      size_t per_adv_size, bool connectable, bt_mgmt_conn_set_cb conn_set_cb,
+		      bt_mgmt_ext_adv_set_cb ext_adv_set_cb)
+{
+	int ret;
+
+	if (adv == NULL) {
+		LOG_ERR("No adv struct receieved");
+		return -EINVAL;
+	}
+
+	if (adv_size == 0) {
+		LOG_ERR("Invalid size of adv struct");
+		return -EINVAL;
+	}
+
+	adv_local = adv;
+	adv_local_size = adv_size;
+	per_adv_local = per_adv;
+	per_adv_local_size = per_adv_size;
+
+	conn_set = conn_set_cb;
+	ext_adv_set = ext_adv_set_cb;
+
+	if (connectable) {
+		ret = bt_le_ext_adv_create(LE_AUDIO_EXTENDED_ADV_CONN_NAME, NULL, &ext_adv);
+		if (ret) {
+			LOG_ERR("Failed to create advertising set");
```

Comment:
This error log should be the same as the one below?
E.g. "Failed to create connectable extended advertising set: %d"

Reply:
Done